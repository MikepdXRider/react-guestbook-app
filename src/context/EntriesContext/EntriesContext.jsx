import React, { createContext, useContext, useState } from 'react'


const EntriesContext = createContext();

const EntriesProvider = ({children}) => {
    const [entries, setEntries] = useState([]);

    return(
        <EntriesContext.Provider value={{entries, setEntries}}>
            {children}
        </EntriesContext.Provider>
    )
}

const useEntries = () => {
    const context = useContext(EntriesContext);
    
    if (context === undefined) throw new Error('useEntries must be declared within an EntriesProvider');
    
    return context;
}

export { EntriesProvider, useEntries }