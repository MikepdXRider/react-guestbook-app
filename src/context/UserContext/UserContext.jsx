import React, { createContext, useContext, useState } from 'react'


const UserContext = createContext();

const UserProvider = ({children}) => {
    const [username, setUsername] = useState('');

    return(
        <UserContext.Provider value={{username, setUsername}}>
            {children}
        </UserContext.Provider>
    )
}


// 
// Could move to useAuth.jsx in the hooks folder. 
// 
const useUser = () => {
    const context = useContext(UserContext);

    if (context === undefined) throw new Error('useUser must be declared within an UserProvider');

    return context;
}

export { UserProvider, useUser }