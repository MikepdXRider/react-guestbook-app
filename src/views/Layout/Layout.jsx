import React from 'react'


// This is where we'll pull in the header and footer components. We'll also have a main element which will contain the {children} props. 
export default function Layout({children}) {
    return (
        <div>
            {/* header */}
            {/* main element wraping {children} */}
            {/* footer */}
        </div>
    )
}
