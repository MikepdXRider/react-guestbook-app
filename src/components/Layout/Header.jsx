import React from 'react'
import { useUser } from '../../context/UserContext/UserContext.jsx'

export default function Header() {
    const {username} = useUser();

    const userBool = Boolean(username);

    return (
        <div>
            {
                username
                ? <p>Welcome, {username}!</p>
                : <p>This is a header!</p>
            }
        </div>
    )
}
