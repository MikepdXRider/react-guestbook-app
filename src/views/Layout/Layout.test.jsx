import react from "react";
import { render } from '@testing-library/react';
import { UserProvider } from '../../context/UserContext/UserContext.jsx';
import { EntriesProvider } from '../../context/EntriesContext/EntriesContext.jsx';
import Layout from "./Layout.jsx";

it('renders the home view', () => {
    const {container} = render(
            <UserProvider>
                <EntriesProvider>
                    <Layout />
                </EntriesProvider>
            </UserProvider>
    );

    expect(container).toMatchSnapshot();
});