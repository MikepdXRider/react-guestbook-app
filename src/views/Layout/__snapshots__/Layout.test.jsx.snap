// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home view 1`] = `
<div>
  <div>
    <header>
      <p>
        This is a header!
      </p>
    </header>
    <footer>
      <p>
        This is a footer!
      </p>
    </footer>
  </div>
</div>
`;
